components/encryption/src/io.rs

```diff
@@ -644,4 +644,26 @@ mod tests {
             assert_eq!(plaintext, written);
         }
     }
+
+    #[test]
+    fn test_do_crypter_in_place() {
```

Comment:
This test doesn't seem very related. Could you add coverage for `poll_read` instead?

Reply:
And remove this test, it doesn't touch any new code paths.